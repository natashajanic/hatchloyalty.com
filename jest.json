{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
