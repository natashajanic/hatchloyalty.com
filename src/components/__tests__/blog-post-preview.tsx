import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import Img from 'gatsby-image'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Text from '../text'
import BlogPostPreview, { IBlogPostPreviewProps } from '../blog-post-preview'

describe('BlogPostPreview', () => {
  let props: IBlogPostPreviewProps
  let mountedComponent: ShallowWrapper<any, any, BlogPostPreview>

  beforeEach(() => {
    props = {
      author: 'John Doe',
      date: 'February 25, 2019',
      excerpt: 'This is the excerpt for a test post.',
      featuredImage: undefined,
      slug: '/2019-02-25-test-post/',
      tags: ['Test', 'Press Release'],
      title: 'Test Post',
    }
    mountedComponent = shallow(<BlogPostPreview {...props} />)
  })

  describe('render', () => {
    it('renders the blog title as a link to the blog post', async () => {
      expect(mountedComponent.find(Link).length).toEqual(5)
      const titleLink = mountedComponent.find(Link).at(0)
      expect(titleLink.text()).toEqual(props.title)
      expect(titleLink.prop('to')).toEqual(`/blog/2019-02-25-test-post/`)
    })

    it('renders the author as a link to their profile page', async () => {
      expect(mountedComponent.find(Link).length).toEqual(5)
      const authorLink = mountedComponent.find(Link).at(1)
      expect(authorLink.text()).toEqual(props.author)
      expect(authorLink.prop('to')).toEqual(`/team/john-doe`)
    })

    it('renders the publication date for the blog post', async () => {
      expect(mountedComponent.text()).toContain('on February 25, 2019')
    })

    it('renders each tag as a link', async () => {
      expect(mountedComponent.find(Link).length).toEqual(5)
      const testTagLink = mountedComponent.find(Link).at(2)
      expect(testTagLink.text()).toEqual('Test')
      expect(testTagLink.prop('to')).toEqual(`/blog/tags/test`)
      const prTagLink = mountedComponent.find(Link).at(3)
      expect(prTagLink.text()).toEqual('Press Release')
      expect(prTagLink.prop('to')).toEqual(`/blog/tags/press-release`)
    })

    it('renders a feature image', async () => {
      mountedComponent.setProps({
        featuredImage: {
          childImageSharp: {
            fluid: {
              aspectRatio: 1,
              src: 'test/imgs/test-blog.png',
              srcSet: '',
              sizes: '',
            },
          },
        },
      })
      expect(mountedComponent.find(Img).length).toEqual(1)
    })

    it('does not render a feature image if none is specified', async () => {
      expect(mountedComponent.find(Img).length).toEqual(0)
    })

    it('renders the excerpt for the blog post', async () => {
      expect(mountedComponent.find(Text).length).toEqual(1)
      expect(
        mountedComponent
          .find(Text)
          .at(0)
          .render()
          .text()
      ).toContain(props.excerpt)
    })

    it('renders an explicit link to the blog post', async () => {
      expect(mountedComponent.find(Link).length).toEqual(5)
      const readMoreLink = mountedComponent.find(Link).last()
      expect(readMoreLink.text()).toEqual('Read More')
      expect(readMoreLink.prop('to')).toEqual(`/blog/2019-02-25-test-post/`)
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
