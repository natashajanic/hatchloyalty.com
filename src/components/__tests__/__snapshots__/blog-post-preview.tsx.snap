// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostPreview render matches the expected snapshot 1`] = `
<Styled(Component)
  blacklist={
    Array [
      "css",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "cursor",
    ]
  }
  border="1px solid"
  borderColor="grayLight"
  borderRadius={4}
  is={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "Body": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": false,
          "rules": Array [
            [Function],
            "max-height: 415px; overflow: scroll;",
          ],
        },
        "defaultProps": Object {
          "blacklist": Array [
            "css",
          ],
          "is": "div",
        },
        "displayName": "Styled(Component)",
        "foldedComponentIds": Array [],
        "propTypes": Object {},
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Footer": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": false,
          "rules": Array [
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "defaultProps": Object {
          "alignItems": "center",
          "blacklist": Array [
            "css",
            "alignItems",
            "border",
            "borderTop",
            "borderRight",
            "borderBottom",
            "borderLeft",
            "borderColor",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
          ],
          "borderColor": "grayLight",
          "borderTop": 1,
          "display": "flex",
          "is": "footer",
          "justifyContent": "space-between",
          "mt": 4,
          "mx": -3,
          "pt": 3,
          "px": 3,
        },
        "displayName": "Panel.Footer",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignItems": [Function],
          "border": [Function],
          "borderBottom": [Function],
          "borderColor": [Function],
          "borderLeft": [Function],
          "borderRight": [Function],
          "borderTop": [Function],
          "display": [Function],
          "justifyContent": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Header": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "rules": Array [
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "defaultProps": Object {
          "alignItems": "center",
          "blacklist": Array [
            "css",
            "alignItems",
            "border",
            "borderTop",
            "borderRight",
            "borderBottom",
            "borderLeft",
            "borderColor",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "flex",
            "flexDirection",
          ],
          "borderBottom": 1,
          "borderColor": "grayLight",
          "display": "flex",
          "is": "header",
          "justifyContent": "space-between",
          "mb": 4,
          "mx": -3,
          "pb": 3,
          "px": 3,
        },
        "displayName": "Panel.Header",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignItems": [Function],
          "border": [Function],
          "borderBottom": [Function],
          "borderColor": [Function],
          "borderLeft": [Function],
          "borderRight": [Function],
          "borderTop": [Function],
          "display": [Function],
          "flex": [Function],
          "flexDirection": [Function],
          "justifyContent": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Title": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": false,
          "rules": Array [
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "defaultProps": Object {
          "blacklist": Array [
            "css",
            "color",
            "bg",
            "fontSize",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "alignItems",
            "display",
          ],
          "color": "offBlack",
          "fontSize": 4,
          "is": "h4",
          "m": 0,
        },
        "displayName": "Styled(Component)",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignItems": [Function],
          "bg": [Function],
          "color": [Function],
          "display": [Function],
          "fontSize": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          "box-shadow: #dfdfdf 0 3px 10px;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
        ],
      },
      "defaultProps": Object {
        "blacklist": Array [
          "cursor",
        ],
        "border": "1px solid",
        "borderColor": "grayLight",
        "borderRadius": 4,
        "m": 0,
        "p": 3,
        "width": "100%",
      },
      "displayName": "Panel",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "alignItems": [Function],
        "border": [Function],
        "borderBottom": [Function],
        "borderColor": [Function],
        "borderLeft": [Function],
        "borderRadius": [Function],
        "borderRight": [Function],
        "borderTop": [Function],
        "display": [Function],
        "flex": [Function],
        "flexDirection": [Function],
        "height": [Function],
        "justifyContent": [Function],
        "m": [Function],
        "mb": [Function],
        "ml": [Function],
        "mr": [Function],
        "mt": [Function],
        "mx": [Function],
        "my": [Function],
        "p": [Function],
        "pb": [Function],
        "pl": [Function],
        "position": [Function],
        "pr": [Function],
        "pt": [Function],
        "px": [Function],
        "py": [Function],
        "width": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "systemComponent": true,
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  key="/2019-02-25-test-post/"
  m={0}
  mb={4}
  p={3}
  px={4}
  py={5}
  width="100%"
>
  <ForwardRef
    style={
      Object {
        "color": "#666666",
        "display": "block",
        "textDecoration": "none",
      }
    }
    to="/blog/2019-02-25-test-post/"
  >
    Test Post
  </ForwardRef>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "color",
        "bg",
        "fontSize",
      ]
    }
    color="grayDark"
    fontSize={1}
    is="div"
  >
    <span>
      Post Written by 
      John Doe
       on 
      February 25, 2019
    </span>
    <br />
    <span>
      Tags: 
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
          ]
        }
        is="span"
        key="tagLink-Test"
        mx={1}
      >
        <ForwardRef
          to="/blog/tags/test"
        >
          Test
        </ForwardRef>
      </Styled(Component)>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
          ]
        }
        is="span"
        key="tagLink-Press Release"
        mx={1}
      >
        <ForwardRef
          to="/blog/tags/press-release"
        >
          Press Release
        </ForwardRef>
      </Styled(Component)>
    </span>
  </Styled(Component)>
  <Text
    blacklist={
      Array [
        "css",
        "fontSize",
        "lineHeight",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "textStyle",
        "alignItems",
        "color",
        "bg",
        "display",
        "flex",
        "fontWeight",
        "textAlign",
      ]
    }
    color="grayDark"
    dangerouslySetInnerHTML={
      Object {
        "__html": "This is the excerpt for a test post.",
      }
    }
    fontSize={3}
    is="p"
    lineHeight="1.5"
    m={0}
    textStyle="normal"
  />
  <ForwardRef
    color="grayDark"
    to="/blog/2019-02-25-test-post/"
  >
    Read More
  </ForwardRef>
</Styled(Component)>
`;
